python/lsst/ctrl/bps/htcondor/htcondor_service.py

```diff
@@ -1392,6 +1408,36 @@ def _get_run_summary(job):
     return summary
 
 
+def _get_exit_code_summary(jobs):
+    """Get the exit code summary for a run.
+
+    Parameters
+    ----------
+    jobs : `dict` [`str`, `dict` [`str`, Any]]
+        Mapping HTCondor job id to job information.
+
+    Returns
+    -------
+    summary : `dict` [`str`, `list` [`int`]]
+        Jobs' exit codes per job label.
+    """
+    summary = {}
+    for job_id, job_ad in jobs.items():
+        job_label = job_ad["bps_job_label"]
+        summary.setdefault(job_label, [])
+        try:
+            if job_ad["JobStatus"] in {JobStatus.COMPLETED, JobStatus.HELD}:
+                try:
+                    exit_code = job_ad["ExitSignal"] if job_ad["ExitBySignal"] else job_ad["ExitCode"]
+                except KeyError:
+                    _LOG.debug("Cannot determine exit code for job '%s'", job_id)
+                else:
+                    summary[job_label].append(exit_code)
+        except KeyError:
+            _LOG.debug("Attribute 'JobStatus' not found in classad for job '%s'", job_id)
+    return summary
```

Comment:
_get_exit_code_summary looks like it might be straightforward to write unittest(s) for (assuming I didn't miss an HTCondor call among the dictionary manipulations).

Reply:
Sure, I'll write unit test(s) for this function.